src/service-worker.js

```diff
@@ -31,52 +31,58 @@ self.addEventListener('activate', event => {
 	);
 });
 
+
+/**
+ * Fetch the asset from the network and store it in the cache. 
+ * Fall back to the cache if the user is offline.
+ */
+async function fetchAndCache(request) {
+	const cache = await caches.open(`offline${timestamp}`)
+
+	try {
+		const response = await fetch(request);
+		cache.put(request, response.clone());
+		return response;
+	} catch (err) {
+		const response = await cache.match(request);
+		if (response) return response;
+
+		throw err;
+	}
+}
+
 self.addEventListener('fetch', event => {
 	if (event.request.method !== 'GET' || event.request.headers.has('range')) return;
 
 	const url = new URL(event.request.url);
 
 	// don't try to handle e.g. data: URIs
-	if (!url.protocol.startsWith('http')) return;
-
-	// ignore dev server requests
-	if (url.hostname === self.location.hostname && url.port !== self.location.port) return;
-
-	// always serve static files and bundler-generated assets from cache
-	if (url.host === self.location.host && cached.has(url.pathname)) {
-		event.respondWith(caches.match(event.request));
-		return;
-	}
+	const isHttp = url.protocol.startsWith('http');
+	const isDevServerRequest = url.hostname === self.location.hostname && url.port !== self.location.port;
+	const isStaticAsset = url.host === self.location.host && staticAssets.has(url.pathname);
+	const skipBecauseUncached = event.request.cache === 'only-if-cached' && !isStaticAsset;
 
-	// for pages, you might want to serve a shell `service-worker-index.html` file,
-	// which Sapper has generated for you. It's not right for every
-	// app, but if it's right for yours then uncomment this section
-	/*
-	if (url.origin === self.origin && routes.find(route => route.pattern.test(url.pathname))) {
-		event.respondWith(caches.match('/service-worker-index.html'));
-		return;
-	}
-	*/
+	if (isHttp && !isDevServerRequest && !skipBecauseUncached) {
+		event.respondWith(
+			(async () => {
+				// always serve static files and bundler-generated assets from cache.
+				// if your application has other URLs with data that will never change,
+				// set this variable to true for them and they will only be fetched once.
+				const serveFromCache = isStaticAsset
```

Comment:
I think it's more confusing to follow to have this because you have to check where `serveFromCache` is used. I liked it better when it was inlined because to me it's just an extra layer of indirection